index.js

```diff
@@ -1,12 +1,19 @@
 'use strict';
-module.exports = (arr, comp) => {
-	for (let i = 0; i < arr.length - 1; i++) {
-		const a = arr[i];
-		const b = arr[i + 1];
 
-		if (comp ? comp(a, b) > 0 : a > b) {
+const defaultComparator = (a, b) => a > b ? 1 : 0;
+module.exports = (arr, comp) => {
+	// pass only iterable objects
+	if (!arr || typeof arr[Symbol.iterator] !== 'function') {
+		return false;
+	}
+	comp = comp || defaultComparator;
+	var a = arr[0];
```

Comment:
But code written exclusively for Node 4/6 as I understand it. It therefore rejected [this PR ](https://github.com/sindresorhus/is-array-sorted/pull/1)